### ISSUE TITLE
Failed to get matadata because "Offset is out of bounds"

### ISSUE THREAD

**Issue by mycoboco** [Borewit/music-metadata#38]
Failed to retrieve metadata from the attached file saying "Offset is out of bounds". Maybe the file has an ill-formed fields, but other clients have no problem with it.

[outofbounds.mp3.zip](https://github.com/Borewit/music-metadata/files/1275854/outofbounds.mp3.zip)

**Comment by Borewit** [comment]
Thanks for providing test data!

The MP3 is indeed somewhat corrupt.
````
Sampling rate changed in frame 3697 from 32000 Hz to 44100 Hz
Frame 3697 header expected at byte 3865480, but found at byte 3866331.
Frame 3696 (bytes 3865191-3866331) was 1140 bytes long (expected 289 bytes).

Frame 4581 header expected at byte 4790021, but found at byte 4790023.
Frame 4580 (bytes 4788976-4790023) was 1047 bytes long (expected 1045 bytes).

Frame 4582 header expected at byte 4791068, but found at byte 4792111.
Frame 4581 (bytes 4790023-4792111) was 2088 bytes long (expected 1045 bytes).

Frame 5581 header expected at byte 5835964, but found at byte 5835966.
Frame 5580 (bytes 5834919-5835966) was 1047 bytes long (expected 1045 bytes).

Frame 5582 header expected at byte 5837011, but found at byte 5838054.
Frame 5581 (bytes 5835966-5838054) was 2088 bytes long (expected 1045 bytes).

Frame 5707 header expected at byte 5968666, but found at byte 5968668.
Frame 5706 (bytes 5967621-5968668) was 1047 bytes long (expected 1045 bytes).

Sampling rate changed in frame 5709 from 44100 Hz to 48000 Hz
Layer changed in frame 5709 from Layer III to Layer I
Frame 5709 header expected at byte 5970758, but found at byte 5971076.
Frame 5708 (bytes 5969713-5971076) was 1363 bytes long (expected 1045 bytes).

Sampling rate changed in frame 5710 from 48000 Hz to 44100 Hz
Layer changed in frame 5710 from Layer I to Layer III
Frame 5710 header expected at byte 5971368, but found at byte 5971801.
Frame 5709 (bytes 5971076-5971801) was 725 bytes long (expected 292 bytes).

Frame 7442 header expected at byte 7781564, but found at byte 7781566.
Frame 7441 (bytes 7780519-7781566) was 1047 bytes long (expected 1045 bytes).

Frame 7443 header expected at byte 7782611, but found at byte 7783654.
Frame 7442 (bytes 7781566-7783654) was 2088 bytes long (expected 1045 bytes).


Summary:
===============
Total number of frames: 8166, unpadded: 834, padded: 7332
File is VBR. Average bitrate is 319 kbps.
Exact length: 03:33
````

**Comment by Borewit** [comment]
Fixed in [v0.8.3](https://github.com/Borewit/music-metadata/releases/tag/v0.8.3)
